internal/grid/connection.go

```diff
@@ -174,9 +174,9 @@ func (c ContextDialer) DialContext(ctx context.Context, network, address string)
 }
 
 const (
-	defaultOutQueue    = 10000
-	readBufferSize     = 16 << 10
-	writeBufferSize    = 16 << 10
+	defaultOutQueue    = 100000
```

Comment:
```suggestion
	defaultOutQueue    = 10000
```

This is per remote. I don't think it will make any significant difference - but just take more memory.

Bigger buffers are fine, though.

You are welcome to try slightly larger values of `maxMergeMessage` if you are indeed seeing the queue filled up: https://github.com/minio/minio/blob/960d6040136ef64ab9b2fc6f34d925646f5d206f/internal/grid/grid.go#L48

In my benchmarks 20 seemed to be the sweet spot - but maybe with double the buffer doubling this could also help.